api4/cloud.go

```diff
@@ -40,6 +40,9 @@ func (api *API) InitCloud() {
 	api.BaseRoutes.Cloud.Handle("/subscription/invoices/{invoice_id:in_[A-Za-z0-9]+}/pdf", api.APISessionRequired(getSubscriptionInvoicePDF)).Methods("GET")
 	api.BaseRoutes.Cloud.Handle("/subscription", api.APISessionRequired(changeSubscription)).Methods("PUT")
 
+	// GET /api/v4/cloud/request-trial
+	api.BaseRoutes.Cloud.Handle("/request-trial", api.APISessionRequired(requestTrial)).Methods("GET")
```

Comment:
Just noticed while reviewing the webapp PR that this route is currently a `GET`. Even though there's no body in the request, `GET` are typically used for requests that don't change the state of the system. Since this is altering the subscription by adding a trial to it, could we instead make this a `PUT` (and update the webapp client to also be a `PUT`)?

Reply:
now that you mention that, yes, I agree that there is actually a change in the state of the license. Changing it. Thanks for the comment!